### ISSUE TITLE
Chromecast issue with YouTube TV

### ISSUE THREAD

**Issue by LunaFennec** [abrenoch/hyperion-android-grabber#211]
Grabber works fine in UI and things like plex and spotify clips... however in youtube TV it seems to only show 3-4 frames before looping them and becoming stuck. This only happens as the mobile release youtube sideloaded on the chromecast works fine. 
When the frames are looping the ui still shows fine through the hyperion live view- however the media itself is still looping.

**Comment by CommanderRedYT** [comment]
Have the same problem on the Chromecast 4K with the Jellyfin for Android TV app

**Comment by CommanderRedYT** [comment]
After some research it seems that the problem exists when there is a video player, and then only for this. For me, the youtube player bar is updating, but the video in the background is looping.

**Comment by CommanderRedYT** [comment]
Okay, I did even more research alongside managing to compile this in Intellij Idea. 

I added some code that computes the hash of the buffer that is being sent and for 500 samples, there are about 9 unique hashes when the problem occurs, meaning the send function already is getting bad data. Thing is, this is called basically directly by this: https://developer.android.com/reference/android/media/ImageReader.OnImageAvailableListener

**Comment by Anashost** [comment]
anybody managed to make it work? mine not working in any player, as explained here: https://github.com/abrenoch/hyperion-android-grabber/issues/208#issue-2677152312

**Comment by CommanderRedYT** [comment]
My setup right now is doing it with hyper-tizen, although that one has HDR issues when the source is HDMI. So it's fine for a game console (where this project can't help anyways) but for watching movies on this it would probably be better also installing it on the Chromecast.

I once tried running the grabber in an emulated tv on my laptop, there it worked fine. However, that one had much newer android version available. (I think it was >=12 or even >=14)

**Comment by Mortirreke** [comment]
> My setup right now is doing it with hyper-tizen, although that one has HDR issues when the source is HDMI. So it's fine for a game console (where this project can't help anyways) but for watching movies on this it would probably be better also installing it on the Chromecast.
> 
> I once tried running the grabber in an emulated tv on my laptop, there it worked fine. However, that one had much newer android version available. (I think it was >=12 or even >=14)

Does this solution also capture the HDMI input of my android TV?

**Comment by CommanderRedYT** [comment]
@Mortirreke unfortunately I have no idea. I just have a Samsung smart tv 2024 and a Chromecast 4K, with those things I can help testing and debugging, but with everything else I can't help with

**Comment by clunkcc** [comment]
I spent a long time last night trying to get this to work, and I thought I'd chime in with the workarounds that I've found that allow the app to capture video and send it to Hyperion.
My setup is running version 1.1-TV on a Google TV with Chromecast running Android 12

To watch Youtube videos, I use SmartTube
To stream TV/Movies, I use Stremio (but Jellyfin would probably work too). I have it set to use an external player
The external player I use is SPMC (an outdated fork of Kodi)
In SPMC, in Settings > Video > Acceleration (make sure you have Settings level = Advanced), I have allowed hardware acceleration - MediaCodec, and Accelerate MPEG2,MPEG4,h264,hevc: Never

This solution isn't very elegant, and there are probably better apps or more optimized settings that I'd be super keen to hear about, but this is a pretty good start

**Comment by Anashost** [comment]
Thanks @alexdobo for sharing!
This is kinda similar to the workaround I use with Jellyfin and IPTV apps that support external players. I use [MPV for Android](https://github.com/mpv-android/mpv-android) as the the external player.

The version that is compatible with Chromecast with Android TV: `app-api29-universal-release.apk`

To get Hyperion grabber to work with it, you'll need to make a small tweak:
Go to MPV Settings > General > disable Hardware decoding.

MPV also offers advanced customization through a config file, which can be really helpful for more fine-tuned control.

By the way, you mentioned you're using SmartTube.. does the grabber work with that app for you? If so, how did you get it to work?

**Comment by artemis-neks** [comment]
In SmartTube go to Settings - Player - Developer options - Force SW video decoder

**Comment by Anashost** [comment]
> In SmartTube go to Settings - Player - Developer options - Force SW video decoder

Thank it works, but made video playback very slow/laggy..

**Comment by artemis-neks** [comment]
> > In SmartTube go to Settings - Player - Developer options - Force SW video decoder
> 
> Thank it works, but made video playback very slow/laggy..

Yep. Buuut... found another solution. 
Settings - Player - Developer options - Ambilight/ aspect ratio/ video scale/ screenshots fix

**Comment by McMoobud** [comment]
I'm having a similar issue out of the blue this week. Setup has worked flawlessly for 1-2 years and now all of a sudden died.

Setup Wled (ws2812b) and Esp8266 Hyperion Grabber on Fire Stick 4K Hyperion running on VM

Yesterday out of the blue, all video apps now hang (Kodi, YouTube etc) and the lights go green.

Seems to originate from the Grabber app specifically, as when deactivated video playback returns to normal. I can get videos running ok with lights on Kodi if I disable hardware acceleration but then 4k playback is choppy as hell.

Anyone seeing similar?

**Comment by watermarkhu** [comment]
> I'm having a similar issue out of the blue this week. Setup has worked flawlessly for 1-2 years and now all of a sudden died.
> 
> Setup Wled (ws2812b) and Esp8266 Hyperion Grabber on Fire Stick 4K Hyperion running on VM
> 
> Yesterday out of the blue, all video apps now hang (Kodi, YouTube etc) and the lights go green.
> 
> Seems to originate from the Grabber app specifically, as when deactivated video playback returns to normal. I can get videos running ok with lights on Kodi if I disable hardware acceleration but then 4k playback is choppy as hell.
> 
> Anyone seeing similar?

Yup me too, although started about 6 months ago. For the the grabber app is able to grab for about 1 second before it becomes stuck. So on the setup screen with the rotating colors it will stop rotating after 1 second.
